test/integration/features/profile.feature

```diff
@@ -2,104 +2,229 @@
+      """
+      Profile 'minishift' set as active profile
+      """   
 
-Scenario: User should be able to list 'minishift' as the active profile
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Running		(Active)
-    """
-
-Scenario: User should be able to delete profile 'foo'
-    When executing "minishift profile delete foo --force" succeeds
-    Then stdout should contain
-    """
-    Profile 'foo' deleted successfully
-    """
-
-Scenario: User can not delete default profile 'minishift'
-    When executing "minishift profile delete minishift --force"
-    Then exitcode should equal "1"
-     And stderr should contain
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given profile "minishift" has state "Does Not Exist"
+     And executing "minishift profile set minishift" succeeds
+     And stdout should contain
      """
-     Default profile 'minishift' can not be deleted
+     Profile 'minishift' set as active profile
      """
+     When executing "minishift start" succeeds
+      Then profile "minishift" should have state "Running"
+       And executing "minishift profile list" succeeds
+       And stdout should match "-\s*minishift\s*Running\s*\(Active\)"
 
-Scenario: Deleting Minishift
-   Given Minishift has state "Running"
-    When executing "minishift delete --force" succeeds
-    Then Minishift should have state "Does Not Exist"
-    When executing "minishift ip"
-    Then exitcode should equal "1"
+  Scenario: As user, I can apply independent settings in profile 'foo'
+    Given profile "foo" has state "Does Not Exist"
+     And executing "minishift profile set foo" succeeds
+     And stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift config set memory 4096" succeeds
+      And executing "minishift config set disk-size 30g" succeeds
+      And executing "minishift config set cpus 4" succeeds
+      And executing "minishift addons enable registry-route" succeeds
+     Then executing "minishift config get memory" succeeds
+      And stdout should match "4096"
+     Then executing "minishift config get disk-size" succeeds
+      And stdout should match "30g"
+     Then executing "minishift config get cpus" succeeds
+      And stdout should match "4"
+     Then executing "minishift addons list"
+      And stdout should match "registry-route\s*: enabled\s*P\(0\)"
 
-Scenario: User should be able to switch between non existing profiles
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Does Not Exist	(Active)
-    """
-    When executing "minishift profile set abc" succeeds
-      Then stdout should contain
-      """
-      Profile 'abc' set as active profile.
+  @minishift-only
+  Scenario: As user, I can check the config settings and add-ons list that should not be changed in profile 'minishift' from active profile 'foo' 
```

Comment:
If we sacrifice the memory check, which is not really crucial, and move the addon check to the individual scenario, then this scenario could be without `@minishift-only`.